lib/wallet.js

```diff
@@ -124,6 +130,156 @@ class Wallet extends ethers.Wallet {
 
         return [approvalTxReceipt, depositTxReceipt];
     }
+
+    /**
+     * Start a payment challenge period for a receipt
+     * @param {Receipt} receipt - The receipt object for payment challenge.
+     * @param {string} stageAmount - The amount to stage from the settled balance during the settlement.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.startChallengeFromPayment(receipt, '1.1', {gasLimit: 200000});
+     */
+    async startChallengeFromPayment(receipt, stageAmount, options = {}) {
+        //if previous challenge phrase is in dispute, not allow to start new challenge
+        const currentPhase = await this.getCurrentPaymentChallengePhase();
+        if (currentPhase === 'Dispute') 
+            throw new Error('Current challenge phase is in dispute, can not start new payment challenge!');
+        
+        
+        let amountBN = ethers.utils.parseEther(stageAmount);
+        const receiptJSON = receipt.toJSON();
+        if (receiptJSON.currency.ct !== '0x0000000000000000000000000000000000000000') {
+            const tokenInfo = await getTokenInfo.call(this, receiptJSON.currency.ct, true);
+            amountBN = ethers.utils.parseUnits(stageAmount.toString(), tokenInfo.decimals);
+        }
+        
+        const challengeContract = _driipSettlementChallengeContract.get(this);
+        return await challengeContract.startChallengeFromPayment(receipt.toJSON(), this.address, amountBN, options);
+    }
+
+    /**
+     * Settle a payment driip of this wallet.
+     * @param {Receipt} receipt - The receipt object for payment challenge.
+     * @param [options]
+     * @returns {Promise} A promise that resolves into transaction hash.
+     * @example
+     * let hashObj = await wallet.settleDriipAsPayment(receipt, {gasLimit: 200000});
+     */
+    async settleDriipAsPayment(receipt, options = {}) {
+        //if current challenge phrase is in dispute or status is disqualified, not allow to settle
+        const currentPhase = await this.getCurrentPaymentChallengePhase();
+        const currentStatus = await this.getCurrentPaymentChallengeStatus();
```

Comment:
What if there are multiple challenges pending from the same wallet? Is that possible?